src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/Gang.kt

```diff
@@ -44,8 +44,19 @@ class Gang(
   @OneToMany(mappedBy = "gang", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
   val members: MutableList<GangMember> = mutableListOf(),
 
+  @OneToMany(mappedBy = "primaryGang", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
+  val nonAssociationsPrimary: List<GangNonAssociation> = listOf(),
+
+  @OneToMany(mappedBy = "secondaryGang", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
+  val nonAssociationsSecondary: List<GangNonAssociation> = listOf(),
+
 ) {
 
+  fun getNonAssociations(): List<Pair<Gang, NonAssociationReason>> =
```

Comment:
This returns a list of Gangs and that should not be associated with the current gang and the reason for the non-association